File: trivially/src/main/java/trivially/ep/m7i2/finalized/Sub.java
package trivially.ep.m7i2.finalized;

import trivially.ep.m7i2.Exp;

public class Sub implements trivially.ep.m7i2.Sub {

    protected Exp left;

    public Exp getLeft() {
        return this.left;
    }

    protected Exp right;

    public Exp getRight() {
        return this.right;
    }

    // Warning: Unsafe casts
    public void setLeft(trivially.ep.m4.Exp left) { this.left = (trivially.ep.m7i2.Exp) left; }
    public void setRight(trivially.ep.m4.Exp right) { this.right = (trivially.ep.m7i2.Exp) right; }

    public Sub(Exp left, Exp right) {
        this.left = left;
        this.right = right;
    }
}

File: coco_csharp/CoCo/EP/M6/Finalized/Sub.cs
namespace EP.M6.Finalized {
    public class Sub : Exp, EP.M6.Sub<Exp> {
        private EP.Exp<Exp> left;
        private EP.Exp<Exp> right;

        public Sub(EP.Exp<Exp> _left, EP.Exp<Exp> _right) {
            this.left = _left;
            this.right = _right;
        }

        public EP.Exp<Exp> getLeft() { return this.left; }
        public void setLeft(EP.Exp<Exp> _left) { this.left = _left; }

        public EP.Exp<Exp> getRight() { return this.right; }
        public void setRight(EP.Exp<Exp> _right) { this.right = _right; }
    }
}